cpp/src/arrow/engine/substrait/extension_set.h

```diff
@@ -226,14 +238,14 @@ class ARROW_ENGINE_EXPORT ExtensionSet {
  private:
   ExtensionIdRegistry* registry_;
   /// The subset of extension registry URIs referenced by this extension set
-  std::vector<util::string_view> uris_;
-  std::vector<TypeRecord> types_;
-
-  std::vector<FunctionRecord> functions_;
-
-  // pimpl pattern to hide lookup details
-  struct Impl;
-  std::unique_ptr<Impl, void (*)(Impl*)> impl_;
+  std::unordered_map<uint32_t, util::string_view> uris_;
+  std::unordered_map<uint32_t, TypeRecord> types_;
+  std::unordered_map<uint32_t, FunctionRecord> functions_;
+  std::unordered_map<Id, uint32_t, IdHashEq, IdHashEq> types_map_, functions_map_;
```

Comment:
Can we add some comments explaining these fields?  Also, combining declarations on one line is discouraged as it can be a bit tricky to read.
```suggestion
  // Map from anchor values to URI values
  std::unordered_map<uint32_t, util::string_view> uris_;
  // Map from anchor values to type definitions, used during Substrait->Arrow
  // and populated from the Substrait extension set
  std::unordered_map<uint32_t, TypeRecord> types_;
  // Map from anchor values to function definitions, used during Substrait->Arrow
  // and populated from the Substrait extension set
  std::unordered_map<uint32_t, FunctionRecord> functions_;
  // Map from type names to anchor values.  Used during Arrow->Substrait
  // and built as the plan is created.
  std::unordered_map<Id, uint32_t, IdHashEq, IdHashEq> types_map_;
  // Map from function names to anchor values.  Used during Arrow->Substrait
  // and built as the plan is created.
  std::unordered_map<Id, uint32_t, IdHashEq, IdHashEq> functions_map_;
```

Reply:
Made the change, thanks!